app/services/delete_content_item.rb

```diff
@@ -0,0 +1,40 @@
+module Services
+  class DeleteContentItem
+    def self.destroy_content_items_with_links(content_ids)
+      content_ids = Array(content_ids)
+
+      content_items = ContentItem.where(content_id: content_ids)
+      destroy_supporting_objects(content_items)
+      content_items.destroy_all
+
+      destroy_links(content_ids)
+    end
+
+    def self.destroy_supporting_objects(content_items)
+      content_items = Array(content_items)
+
+      supporting_classes = [
+        AccessLimit,
+        Location,
+        State,
+        Translation,
+        Unpublishing,
+        UserFacingVersion,
+        ChangeNote,
+      ]
+
+      supporting_classes.each do |klass|
+        next unless ActiveRecord::Base.connection.data_source_exists?(klass.table_name)
+        klass.where(content_item: content_items).destroy_all
+      end
+
+      LockVersion.where(target: content_items).destroy_all
+    end
+
+    def self.destroy_links(content_ids)
+      content_ids = Array(content_ids)
+      LinkSet.where(content_id: content_ids).destroy_all
+      Link.where(target_content_id: content_ids).destroy_all
```

Comment:
Good spot :) This code is just being moved in this branch (from here: [1]), so you might want to make a separate Pull Request for this change?

1: https://github.com/alphagov/publishing-api/blob/c183b846d90c254953a81f049ed83f3e31ac272e/db/migrate/helpers/delete_content_item.rb#L37